src/mds/Server.cc

```diff
@@ -9241,7 +9241,9 @@ bool Server::_dir_has_snaps(const MDRequestRef& mdr, CInode *diri)
   ceph_assert(diri->snaplock.can_read(mdr->get_client()));
 
   SnapRealm *realm = diri->find_snaprealm();
-  return !realm->get_snaps().empty();
+  auto& snaps = realm->get_snaps();
+  auto it = snaps.lower_bound(diri->get_oldest_snap());
+  return it != snaps.end();
```

Comment:
nit: I find following more succinct -
```suggestion
  return snaps.lower_bound(diri->get_oldest_snap()) != snaps.end();
```